JH7110 fixes

orangecms opened the pull request.

orangecms committed:
dts: jh7110: fix memory definitions

The L2LIM is 2MB in size, see:
https://doc-en.rvspace.org/JH7110/TRM/JH7110_TRM/u74_memory_map.html
Rename it since there are other memory blocks such as the DTIM for the S7.

Signed-off-by: Daniel Maslowski <info@orangecms.org>

orangecms committed:
soc: starfive: jh71xx: select 64BIT for SOC_JH7110

We get a 32bit binary otherwise.

Signed-off-by: Daniel Maslowski <info@orangecms.org>

ycsin commented:
cc @pfarwsi  @kanakshilledar

pfarwsi commented on dts/riscv/starfive/jh7110-visionfive-v2.dtsi:
```diff
@@ -133,9 +133,14 @@
+
+	dtim: memory@01100000 {
+		device_type = "memory";
+		reg = <0x0 0x01100000 0x0 0x2000>;
```
Did you mean `S7 Hart0 DTIM` here? As per the JH7110 TRM here: [https://doc-en.rvspace.org/JH7110/TRM/JH7110_TRM/u74_memory_map.html](https://doc-en.rvspace.org/JH7110/TRM/JH7110_TRM/u74_memory_map.html) the address space starts at `0x00_0110_1000`.

orangecms replied:
Yes and that is an erratum in the TRM.

From reverse engineering the mask ROM and reading out the DTIM, I know that it really starts at the address I wrote here. Then the size in the TRM also fits with the end address.

pfarwsi replied:
Agreed on the TRM errata, needs correction on starfive's part. This is from sifive's official books: [https://starfivetech.com/uploads/u74mc_core_complex_manual_21G1.pdf](https://starfivetech.com/uploads/u74mc_core_complex_manual_21G1.pdf), Page number `96` says:  

> 0x0100_0000 0x0100_1FFF RWX A S7 Hart 0 DTIM (8 KiB)

Still not the address it mentions here and also you intend to mention.

orangecms replied:
I would think that's the reference design.

All I can say is that I dumped the memory area I put in here and it had the expected data. You can do that as well to verify, e.g. by adding some code to oreboot.

kanakshilledar replied:
We are still trying to understand the significance of this `dtim` node in the zephyr context here.

Provide some steps to verify the **S7 Hart 0 DTIM** address.

orangecms replied:
There are two ways coming to my mind:
1. dump and reverse engineer the mask ROM, e.g. via U-Boot + TFTP
2. read out from the address I provide at runtime, e.g. via oreboot or U-Boot

kanakshilledar commented on dts/riscv/starfive/jh7110-visionfive-v2.dtsi:
```diff
@@ -133,9 +133,14 @@
+		reg = <0x0 0x8000000 0x0 0x200000>;
+	};
+
+	dtim: memory@01100000 {
```
Remove the leading zero in memory address. 
It gives warning during the build process.
```suggestion
	dtim: memory@1100000 {
```

orangecms replied:
Ah yes, that is annoying. I wanted to use 8 digits to make it easier to read. Fix to come!

ycsin commented:
@orangecms might need a rebase

orangecms commented:
I've omitted the DTIM memory node for now. We will get back to this in another PR. :)

ycsin commented:
Could you please check the title and the body of the "fix memory definitions" commit? Looks a little weird to me

orangecms commented:
> Could you please check the title and the body of the "fix memory definitions" commit? Looks a little weird to me

What do you suggest changing it into?

orangecms commented:
I've changed the message title and added a reference to the memory map / manual, any feedback @ycsin ?

ycsin commented:
> I've changed the message title and added a reference to the memory map / manual, any feedback @ycsin ?

The reference is nice to have, thanks for that

`{board,dts}: JH7110: fix memory definitions` looks unconventional, I'd do `dts: jh7110: fix memory definitions` instead

`soc/starfive/jh71xx/Kconfig: add 64BIT` -> `soc: starfive: jh71xx: select 64BIT for SOC_JH7110` - we usually use `: ` instead of `/`, except for title that starts with `subsys`

orangecms commented:
> > I've changed the message title and added a reference to the memory map / manual, any feedback @ycsin ?
> 
> The reference is nice to have, thanks for that
> 
> `{board,dts}: JH7110: fix memory definitions` looks unconventional, I'd do `dts: jh7110: fix memory definitions` instead
> 
> `soc/starfive/jh71xx/Kconfig: add 64BIT` -> `soc: starfive: jh71xx: select 64BIT for SOC_JH7110` - we usually use `: ` instead of `/`, except for title that starts with `subsys`

Okay, gotcha, will do that :)

orangecms commented:
done

kanakshilledar reviewed (commented):
I am fine with the rest of the changes. For passing the compliance can you please update the commit message to not have more than 75 characters in one line (here 76). For the `dts: jh7110: fix memory definitions commit`

pfarwsi commented:
LGTM @ycsin

orangecms commented:
> I am fine with the rest of the changes. For passing the compliance can you please update the commit message to not have more than 75 characters in one line (here 76). For the `dts: jh7110: fix memory definitions commit`

Oh gosh... I just removed the commas. Pesky picky rules. xD

kanakshilledar reviewed (approved):
LGTM @ycsin
